### ISSUE TITLE
Add dtype for ClassLabel

### ISSUE THREAD

**Issue by EternalMoment** [tensorflow/datasets#619]
**Is your feature request related to a problem? Please describe.**
The default output dtype from `ClassLabel` is `tf.int64`, it would be a overkill for some dataset, such as binary classification problem. Well, the user could cast it with `map` operation afterwards on the fly, but it's better to preprocess the type and store it with the right type.

**Describe the solution you'd like**
let's add a `dtype` arg to `ClassLabel`, with default `tf.int64`. It should allow any `tf.int*` type as input.

**Describe alternatives you've considered**
Well, just let the user cast `tf.int64` on the fly if he/she needs other type .

**Additional context**
It should be a simple change, I could raise a PR

**Comment by Conchylicultor** [comment]
Thanks for reporting, but I'm not sure what is the issue with having `tf.int64` ? Intuitively, I would say the overhead is negligible vs the actual network computation and the image decoding.

Before doing something like this, I would prefer to see some benchmark which demonstrate that using `tf.int64` is indeed a speed/size limitation.

Additionally:
* The internal proto format `tf.train.Example` only support tf.int64, tf.string and tf.float32 as data, so it wouldn't change the internal representation
* It would be more consistent for the user that all labels always have the same dtype.
* This may not be the case anymore but due to historical reason `tf.int64` is better supported on GPU than `tf.int32`

**Comment by EternalMoment** [comment]
I see, I wasn't aware of the underlying proto format tf.train.Example supports `tf.int64` only. In that case, this proposal doesn't make sense. Let me close it.
